packages/project-account-switcher/src/presenters/ProjectAccountSwitcherPresenter.js

```diff
@@ -0,0 +1,224 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import Flyout from "@hig/flyout";
+import "@hig/flyout/build/index.css";
+import icons from "../../../icons/build";
+import { anchorPoints } from "../../../flyout/src/anchorPoints";
+import "./ProjectAccountSwitcherPresenter.scss";
+
+export default class ProjectAccountSwitcherPresenter extends Component {
+  static constructPlaceholder(label) {
+    return label.match(/\b(\w)/g).join("");
+  }
+
+  static propTypes = {
+    /** Heading title for the list of Accounts */
+    accountTitle: PropTypes.string,
+    /** List of Accounts */
+    accounts: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string,
+        image: PropTypes.string,
+        label: PropTypes.string
+      })
+    ),
+    /** Currently selected Account */
+    activeAccountId: PropTypes.string,
+    /** Label for selected Accounts and Projects */
+    activeLabel: PropTypes.string,
+    /** Currently selected Project */
+    activeProjectId: PropTypes.string,
+    /** Handles Account selection, passed from Behavior */
+    onAccountClick: PropTypes.func,
+    /** Handles Project selection, passed from Behavior */
+    onProjectClick: PropTypes.func,
+    /** Handles Flyout display, passed from Behavior */
+    onTargetClick: PropTypes.func,
+    /** Menu heading title for the list of Projects */
+    open: PropTypes.bool,
+    /** Heading title for the list of Projects */
+    projectTitle: PropTypes.string,
+    /** List of Projects */
+    projects: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string,
+        image: PropTypes.string,
+        label: PropTypes.string
+      })
+    )
+  };
+
+  static defaultProps = {
+    accountTitle: "Accounts",
+    accounts: [
+      { id: "1", label: "Account 1" },
```

Comment:
Is it necessary to have default accounts and projects?

Reply:
As the component is currently built, yes I think so. `activeAccount` and `activeProfile` draw from `accounts` and `projects`. If `accounts` and `projects` are removed from the default props, and no props are passed to the component (https://github.com/Autodesk/hig/pull/1102/files#diff-fea449e3e59126168ca3e3e517b50d2cR8), it cannot render. 

Should `accounts` and `projects` always be required to be passed to the component?

Reply:
Yeah, it's good that we're starting to some pre-emptive API design.  

Does the vanilla version have defaults?

In an ideal world I'd probably make one of`accounts` or `projects` required. If one isn't supplied then it renders without that part of the list.  That might be a pain to do now. This is more of a product-y question though.  @nfiniteset Thoughts?